### PR TITLE
reorg cache fix

### PR DESCRIPTION
**Pull request by deevope**
Related to #3489 

- **reorg cache period configurable** :  Allow everyone to configure themselves the reorg cache period, so they could increase it easily in function of their constraint. To repopulate their local mempool.

- ~~**treat cached txs as "new" during reconciliation** :  Allow our nodes to broadcast txs to their peers.~~
>  ~~_For every tx in the reorg cache that is successfully added back to the mempool during reconciliation should also be (re)broadcast out to our peers. We should treat these txs as "new" in this scenario._~~

- ~~**increase max pool size to 300K txs** : To prevent for the future, in case of large reorg with empty blocks, against stale full blocks, nodes mempool should be able to handle 6 hours of transactions of full blocks.~~

- ~~**Evict duplicate txs in reorg cache** : Given we add transactions back transaction to the pool, and in the pool we add them back in the reorg cache, we remove them from the reorg cache before they are sent in the pool.~~

- ~~**Evict only txs accepted to pool and sourced from reorg cache when rec…** : Fix of commit [3d8df3a](https://github.com/mimblewimble/grin/pull/3495/commits/3d8df3a7c82ecd8c0f05c25c87b05a7013957ee4), It will evict _only_ from reorg cache, transactions sourced from reorg cache, accepted by the pool to be broadcasted and added to the reorg_cache again. As we want to keep every transactions that were not accepted by the pool in the reorg_cache in the case where a reorg with larger depth happen again.~~

**test**: 
- ~~reorg on testnet made, node rebroadcasted successfully the transactions to other peers~~

- ~~I plan also to make a stress test on testnet with 50k transactions - not done yet.~~

### CODE REVIEW

**Comment by deevope** [comment]
@antiochp There's one problem on my PR, I'm working to solve it. I should remove from reorg_cache only txs that are sourced from reorg cache and accepted by the pool to be broadcasted again, and as well added back to the reorg cache. Currently it removes every txs in reorg cache once a reorg happen. As we want to keep every transactions that were not accepted by the pool in the reorg_cache for the case where a reorg with larger depth happen again.

**Comment by deevope** [comment]
@antiochp Everything should be fine now and it's been tested on my side. 👍

File: pool/src/types.rs [hunk 1]
```diff
@@ -107,6 +107,11 @@ pub struct PoolConfig {
 	#[serde(default = "default_accept_fee_base")]
 	pub accept_fee_base: u64,
 
+	// Reorg cache retention period in minute.
+	// The reorg cache repopulates local mempool in a reorg scenario.
+	#[serde(default = "default_reorg_cache_period")]
+	pub reorg_cache_period: i64,
```
Comment (antiochp) [inline pool/src/types.rs:113]: Lets make this `u64` as it does not make sense to support `-ve` values in the config.

**Review by antiochp — COMMENTED** [review]
Yes this looks good.
We should investigate the various re-broadcast strategies that have been discussed recently. But exposing this config now makes a lot of sense.

Minor comment on `i64` vs `u64`.

:+1:

File: pool/src/types.rs (hunk 1 above)
Comment (deevope) [reply pool/src/types.rs:113]: I can't do `u64`, as [minutes()](https://docs.rs/chrono/0.4.11/chrono/struct.Duration.html#method.minutes) requires `i64`. Either we set `u32` and turn it to `i64` in minutes() or we just let `i64` but no error will be thrown if users put a negative value.

Comment (antiochp) [reply pool/src/types.rs:113]: `u32` works then, we're not going to need huge values in here.

**Review by antiochp — COMMENTED** [review]
:+1: Looks good.

**Review by antiochp — APPROVED** [review]

**Comment by antiochp** [comment]
Going to merge this for 5.0.0 beta.

**Pull request merged**